x-pack/plugins/synthetics/server/rest_api/synthetics_service/add_monitor_push.ts

```diff
@@ -0,0 +1,370 @@
+    const encryptedSavedObjectsClient = server.encryptedSavedObjects.getClient();
+    const staleMonitorsMap = await getAllPushMonitorsForSuite(savedObjectsClient, projectId);
+    const createdMonitors: string[] = [];
+    const deletedMonitors: string[] = [];
+    const updatedMonitors: string[] = [];
+    const staleMonitors: string[] = [];
+    const failedMonitors: FailedMonitors = [];
+    const failedStaleMonitors: FailedMonitors = [];
+
+    await Promise.all(
+      monitors.map((monitor) =>
+        configurePushMonitor({
+          savedObjectsClient,
+          encryptedSavedObjectsClient,
+          locations,
+          monitor,
+          createdMonitors,
+          updatedMonitors,
+          failedMonitors,
+          server,
+          staleMonitorsMap,
+          projectId,
+        })
+      )
+    );
+
+    await handleStaleMonitors({
+      savedObjectsClient,
+      staleMonitorsMap,
+      staleMonitors,
+      deletedMonitors,
+      server,
+      keepStale,
+      failedStaleMonitors,
+    });
+
+    return response.ok({
+      body: {
+        createdMonitors,
+        updatedMonitors,
+        staleMonitors,
+        deletedMonitors,
+        failedMonitors,
+        failedStaleMonitors,
+      },
+    });
+  },
+});
+
+const getExistingMonitor = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  journeyId: string,
+  projectId: string
+): Promise<SavedObjectsFindResult<EncryptedSyntheticsMonitor>> => {
+  const filter = `${getSuiteFilter(projectId)} AND ${syntheticsMonitorType}.attributes.${
+    ConfigKey.JOURNEY_ID
+  }: "${journeyId}"`;
+  const { saved_objects: savedObjects } = await savedObjectsClient.find<EncryptedSyntheticsMonitor>(
+    {
+      type: syntheticsMonitorType,
+      perPage: 1,
+      filter,
+    }
+  );
+  return savedObjects[0];
+};
+
+const getAllPushMonitorsForSuite = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  projectId: string
+): Promise<StaleMonitorMap> => {
+  const staleMonitors: StaleMonitorMap = {};
+  let page = 1;
+  let totalMonitors = 0;
+  do {
+    const { total, saved_objects: savedObjects } = await getPushMonitorsForSuite(
+      savedObjectsClient,
+      page,
+      projectId
+    );
+    savedObjects.forEach((savedObject) => {
+      const journeyId = (savedObject.attributes as BrowserFields)[ConfigKey.JOURNEY_ID];
+      if (journeyId) {
+        staleMonitors[journeyId] = {
+          stale: true,
+          savedObjectId: savedObject.id,
+          journeyId,
+        };
+      }
+    });
+
+    page++;
+    totalMonitors = total;
+  } while (Object.keys(staleMonitors).length < totalMonitors);
+  return staleMonitors;
+};
+
+const getPushMonitorsForSuite = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  page: number,
+  projectId: string
+) => {
+  return await savedObjectsClient.find<EncryptedSyntheticsMonitor>({
+    type: syntheticsMonitorType,
+    page,
+    perPage: 1,
```

Comment:
not sure why using page size 1?

Reply:
I think this is copypasta from another part of the logic.